### DISCUSSION TITLE
No query results for model [..‚

### DISCUSSION THREAD

**Discussion by infiniteroles** [Laravel-Backpack/community-forum#998] [Q&A (Help)] [answered]
Hi all, I´m getting an error in one of my models, nad I can not find any log about it. When I edit one user, I get this error:
404
Page not found.

No query results for model [App\Models\User].

But I don´t know what´s wrong. 

I have uploaded a video showing the error. I can not finad any error in my laravel.log, 
https://github.com/Laravel-Backpack/community-forum/assets/11851266/be7f3154-c765-4af5-a720-5f97bfa133f6

Thank you very much

**Comment by jcastroa87** [comment]
Hello @infiniteroles 

Can you share the controller and model code. How did you set "APP_ENV" and "APP_DEBUG"?

Cheers.

**Reply by infiniteroles** [reply]
Hi, this is my config:
APP_ENV=local
LOG_LEVEL=debug
APP_LOG=daily
APP_DEBUG=true


model code:
lass User extends Authenticatable implements MustVerifyEmail
{.....
    use .....
    protected $fillable = [
        'id',
        'name',
      //  'referrer',
        'email',
        'onboarding_id',
        'password',
        'kyc_status',
        'user_type_id',
        'fee_setup_cherry',
        'fee_payment_cherry',
        'fee_setup_user',
        'fee_payment_user',
        'contact_permission_level',
        'risk',
        'risk_date',
        'user_profit_sent_at',
        'user_referral_sent_at',
        'two_factor_confirmed_at',
    ];

    /**
     * The attributes that should be hidden for arrays.
     *
     * @var array
     */
    protected $hidden = [
        'password',
        'remember_token',
        'two_factor_recovery_codes',
        'two_factor_secret',
        'role',
    ];
.....

and crud controller:
...
namespace App\Http\Controllers\Admin;

use App\Http\Requests\UserRequest;
use Backpack\CRUD\app\Http\Controllers\CrudController;
use Backpack\CRUD\app\Library\CrudPanel\CrudPanelFacade as CRUD;

/**
 * Class UserCrudController
 * @package App\Http\Controllers\Admin
 * @property-read \Backpack\CRUD\app\Library\CrudPanel\CrudPanel $crud
 */
class UserCrudController extends CrudController
{
    use \Backpack\CRUD\app\Http\Controllers\Operations\ListOperation;
    use \Backpack\CRUD\app\Http\Controllers\Operations\CreateOperation;
    use \Backpack\CRUD\app\Http\Controllers\Operations\UpdateOperation;
    use \Backpack\CRUD\app\Http\Controllers\Operations\DeleteOperation;
    use \Backpack\CRUD\app\Http\Controllers\Operations\ShowOperation;

    /**
     * Configure the CrudPanel object. Apply settings to all operations.
     * 
     * @return void
     */
    public function setup()
    {
        CRUD::setModel(\App\Models\User::class);
        CRUD::setRoute(config('backpack.base.route_prefix') . '/user');
        CRUD::setEntityNameStrings('user', 'users');
    }

    /**
     * Define what happens when the List operation is loaded.
     * 
     * @see  https://backpackforlaravel.com/docs/crud-operation-list-entries
     * @return void
     */
    protected function setupListOperation()
    {
        CRUD::setOperationSetting('lineButtonsAsDropdown', true);
        CRUD::denyAccess('create');
       // CRUD::setFromDb(); // set columns from db columns.
        CRUD::addColumn([
            'name' => 'id',
            'type' => 'text',
            'label' => 'Id',
        ]);
        CRUD::addColumn([
            'name' => 'name',
            'type' => 'text',
            'label' => 'Name',
        ]);
        CRUD::addColumn([
            'name' => 'email',
            'type' => 'email',
            'label' => 'Email',
        ]); 
        CRUD::addColumn([
            'name' => 'kyc_status',
            'type' => 'text',
            'label' => 'KYC Status',
        ]);
  }

    /**
     * Define what happens when the Create operation is loaded.
     * 
     * @see https://backpackforlaravel.com/docs/crud-operation-create
     * @return void
     */
    protected function setupCreateOperation()
    {
        CRUD::setValidation(UserRequest::class);
        //CRUD::setFromDb(); // set fields from db columns.
        CRUD::addField([
            'name' => 'id',
            'type' => 'text',
            'label' => 'Id',
            'attributes' => [
                'readonly'    => 'readonly',
                'disabled'    => 'disabled',
            ]   
        ]);
        CRUD::addField([
            'name' => 'uuid',
            'type' => 'text',
            'label' => 'uuid',
            'attributes' => [
                'readonly'    => 'readonly',
                'disabled'    => 'disabled',
            ]   
        ]);
        CRUD::addField([
            'name' => 'name',
            'type' => 'text',
            'label' => 'Nombre',
        ]);
 }

    /**
     * Define what happens when the Update operation is loaded.
     * 
     * @see https://backpackforlaravel.com/docs/crud-operation-update
     * @return void
     */
    protected function setupUpdateOperation()
    {
        $this->setupCreateOperation();
    }

}




Thanks so much!!!

**Reply by jcastroa87** [reply]
I can not understand fully the code, because the format.

Did you override any query? did you override any API or Web route? Did the record have a deleted_at?

Please provide more information about your CRUD and version of Backpack.

```
php artisan backpack:version
```

Cheers

**Comment by infiniteroles** [comment]
Sorry abut format and thank you very much for your attention, please see this formated information:

**-Versions:**

### LARAVEL VERSION:
10.41.0.0

### BACKPACK PACKAGE VERSIONS:
backpack/basset: 1.2.2
backpack/crud: 6.6.1
backpack/generators: v4.0.3
backpack/logmanager: v5.0.1
backpack/pro: 2.1.2
backpack/theme-tabler: 1.2.0


- We have **deleted_at** column in users database
- I don´t understand what you mean with this question ""Did you override any query? did you override any API or Web route? "


Code: Model:
`<?php

namespace App\Models;

use Backpack\CRUD\app\Models\Traits\CrudTrait;
use App\Observers\UserObserver;
use Illuminate\Contracts\Auth\MustVerifyEmail;
use Illuminate\Database\Eloquent\Builder;
use Illuminate\Database\Eloquent\Factories\HasFactory;
use Illuminate\Database\Eloquent\SoftDeletes;
use Illuminate\Foundation\Auth\User as Authenticatable;
use Illuminate\Notifications\Notifiable;
use Laravel\Fortify\TwoFactorAuthenticatable;
use Laravel\Jetstream\HasProfilePhoto;
use Laravel\Sanctum\HasApiTokens;
use Illuminate\Support\Str;
use Illuminate\Support\Carbon;
use Illuminate\Support\Facades\DB;
use Vinkla\Hashids\Facades\Hashids;
use App\Notifications\RecoverPasswordNotification;
use Illuminate\Support\Facades\Auth; 

class User extends Authenticatable implements MustVerifyEmail
{
    use CrudTrait;
    use HasFactory;
    use HasProfilePhoto;
    use Notifiable;
    use TwoFactorAuthenticatable;
    use UserReferral;
    use UserPromotionalCode;
    use SoftDeletes;

    // Admin Role ID number
    public $adminID = 100;

    /**
     * The attributes that are mass assignable.
     *
     * @var array
     */
    protected $fillable = [
        'id',
        'name',
        'email',
        'onboarding_id',
        'password',
        'kyc_status',
        'user_type_id',
        'fee_setup_cherry',
        'fee_payment_cherry',
        'fee_setup_user',
        'fee_payment_user',
        'contact_permission_level',
        'risk',
        'risk_date',
        'user_profit_sent_at',
        'user_referral_sent_at',
        'two_factor_confirmed_at',
    ];

    /**
     * The attributes that should be hidden for arrays.
     *
     * @var array
     */
    protected $hidden = [
        'password',
        'remember_token',
        'two_factor_recovery_codes',
        'two_factor_secret',
        'role',
    ];






    /**
     * The attributes that should be cast to native types.
     *
     * @var array
     */
    protected $casts = [
        'email_verified_at' => 'datetime',
    ];

    /**
     * The accessors to append to the model's array form.
     *
     * @var array
     */
    protected $appends = [
        'profile_photo_url',
        'referral_link',
        'email_verified_at_pretty',
        'two_factor_status',
    ];
}
`


Code Controller:
`<?php

namespace App\Http\Controllers\Admin;

use App\Http\Requests\UserRequest;
use Backpack\CRUD\app\Http\Controllers\CrudController;
use Backpack\CRUD\app\Library\CrudPanel\CrudPanelFacade as CRUD;

/**
 * Class UserCrudController
 * @package App\Http\Controllers\Admin
 * @property-read \Backpack\CRUD\app\Library\CrudPanel\CrudPanel $crud
 */
class UserCrudController extends CrudController
{
    use \Backpack\CRUD\app\Http\Controllers\Operations\ListOperation;
    use \Backpack\CRUD\app\Http\Controllers\Operations\CreateOperation;
    use \Backpack\CRUD\app\Http\Controllers\Operations\UpdateOperation;
    use \Backpack\CRUD\app\Http\Controllers\Operations\DeleteOperation;
    use \Backpack\CRUD\app\Http\Controllers\Operations\ShowOperation;

    /**
     * Configure the CrudPanel object. Apply settings to all operations.
     * 
     * @return void
     */
    public function setup()
    {
        CRUD::setModel(\App\Models\User::class);
        CRUD::setRoute(config('backpack.base.route_prefix') . '/user');
        CRUD::setEntityNameStrings('user', 'users');
    }

    /**
     * Define what happens when the List operation is loaded.
     * 
     * @see  https://backpackforlaravel.com/docs/crud-operation-list-entries
     * @return void
     */
    protected function setupListOperation()
    {
        CRUD::setOperationSetting('lineButtonsAsDropdown', true);
        CRUD::denyAccess('create');
       // CRUD::setFromDb(); // set columns from db columns.
        CRUD::addColumn([
            'name' => 'id',
            'type' => 'text',
            'label' => 'Id',
        ]);
        CRUD::addColumn([
            'name' => 'name',
            'type' => 'text',
            'label' => 'Name',
        ]);
        CRUD::addColumn([
            'name' => 'email',
            'type' => 'email',
            'label' => 'Email',
        ]); 
        CRUD::addColumn([
            'name' => 'kyc_status',
            'type' => 'text',
            'label' => 'KYC Status',
        ]);
/*         CRUD::addColumn([
            'name' => 'user_type_id',
            'type' => 'text',
            'label' => 'User Type',
        ]); */
        CRUD::addColumn([
            'name' => 'onboarding_id',
            'type' => 'text',
            'label' => 'Id Onboardin',
        ]);
/*         CRUD::addColumn([
            'name' => 'uuid',
            'type' => 'text',
            'label' => 'UUID',
        ]); */
/*         CRUD::addColumn([
            'name' => 'contact_permission_level',
            'type' => 'text',
            'label' => 'Contact Permission Level',
        ]);
        CRUD::addColumn([
            'name' => 'risk',
            'type' => 'text',
            'label' => 'Risk',
        ]); */
        CRUD::addColumn([
            'name' => 'created_at',
            'type' => 'datetime',
            'label' => 'Created At',
        ]);
        CRUD::filter('kyc_status')
        ->type('text')
        ->label('STATUS')
        ->whenActive(function($value) {
            CRUD::addClause('where', 'kyc_status', 'LIKE', '%'.$value.'%');
        })->else(function() {
            // nada
        });
        CRUD::filter('name')
        ->type('text')
        ->label('NOMBRE')
        ->whenActive(function($value) {
            CRUD::addClause('where', 'name', 'LIKE', '%'.$value.'%');
        })->else(function() {
            // nada
        });
        CRUD::filter('email')
        ->type('text')
        ->label('EMAIL')
        ->whenActive(function($value) {
            CRUD::addClause('where', 'email', 'LIKE', '%'.$value.'%');
        })->else(function() {
            // nada
        });
        /**
         * Columns can be defined using the fluent syntax:
         * - CRUD::column('price')->type('number');
         */

        CRUD::addButtonFromModelFunction('line', 'download_media', 'downloadMedia', 'end');
        CRUD::addButtonFromModelFunction('line', 'open_alice', 'openAlice', 'end');
        CRUD::addButtonFromModelFunction('line', 'open_dash', 'openDash', 'end');
        CRUD::addButtonFromModelFunction('line', 'open_wallet', 'openWallet', 'end');
        CRUD::addButtonFromModelFunction('line', 'emulate_user', 'emulateUser', 'end');
    }

    /**
     * Define what happens when the Create operation is loaded.
     * 
     * @see https://backpackforlaravel.com/docs/crud-operation-create
     * @return void
     */
    protected function setupCreateOperation()
    {
        CRUD::setValidation(UserRequest::class);
        //CRUD::setFromDb(); // set fields from db columns.
        CRUD::addField([
            'name' => 'id',
            'type' => 'text',
            'label' => 'Id',
            'attributes' => [
                'readonly'    => 'readonly',
                'disabled'    => 'disabled',
            ]   
        ]);
        CRUD::addField([
            'name' => 'uuid',
            'type' => 'text',
            'label' => 'uuid',
            'attributes' => [
                'readonly'    => 'readonly',
                'disabled'    => 'disabled',
            ]   
        ]);
        CRUD::addField([
            'name' => 'name',
            'type' => 'text',
            'label' => 'Nombre',
        ]);
        /* CRUD::addField([
            'name' => 'email',
            'type' => 'email',
            'label' => 'Email',
        ]);
        CRUD::addField([
            'name' => 'kyc_status',
            'type' => 'text',
            'label' => 'Estado KYC',
            'attributes' => [
                'readonly'    => 'readonly',
                'disabled'    => 'disabled',
            ]   
        ]);

        CRUD::addField([
            'name' => 'onboarding_id',
            'type' => 'text',
            'label' => 'Id Onboarding',
            'attributes' => [
                'readonly'    => 'readonly',
                'disabled'    => 'disabled',
            ]   
        ]);

        CRUD::addField([
            'name' => 'contact_permission_level',
            'type' => 'text',
            'label' => 'Nivel de comunicación cliente (0,1,2,3)',
        ]);
        CRUD::addField([
            'name' => 'risk',
            'type' => 'select_from_array',
            'label' => 'Risk',
            'options' => [
                'ALTO' => 'ALTO',
                'MEDIO' => 'MEDIO',
                'BAJO' => 'BAJO',
            ],
        ]);
        CRUD::addField([
            'name' => 'fee_setup_cherry',
            'type' => 'text',
            'label' => 'Fee Setup Cherry',
        ]);
        CRUD::addField([
            'name' => 'fee_payment_cherry',
            'type' => 'text',
            'label' => 'Fee Payment Cherry',
        ]);
        CRUD::addField([
            'name' => 'fee_setup_user',
            'type' => 'text',
            'label' => 'Fee Setup User',
        ]);
        CRUD::addField([
            'name' => 'fee_payment_user',
            'type' => 'text',
            'label' => 'Fee From Referals',
        ]);
        CRUD::addField([
            'name' => 'risk_date',
            'type' => 'datetime',
            'label' => 'Risk Date',
        ]); */
        /**
         * Fields can be defined using the fluent syntax:
         * - CRUD::field('price')->type('number');
         */
    }

    /**
     * Define what happens when the Update operation is loaded.
     * 
     * @see https://backpackforlaravel.com/docs/crud-operation-update
     * @return void
     */
    protected function setupUpdateOperation()
    {
        $this->setupCreateOperation();
    }

}
`

**Reply by jcastroa87** [reply]
Is look like a very customized CRUD, to be honest if i didn't have access to the full code to replicate in my side will be hard to find a solution, i can recommend to you start with a clean Controller, because look like one of the many import you are using override the query on create or select. Thats could be the reason why after save didnt work.

If you start with a clean Controller who work normal, add the import one by one to find the one who is failing.

Cheers.

**Reply by munjaldevelopment** [reply, marked as answer]
Hi @infiniteroles ,

As I understood the code at glance, I think this line is causing error:

`CRUD::addField([
'name' => 'id',
'type' => 'text',
'label' => 'Id',
'attributes' => [
'readonly' => 'readonly',
'disabled' => 'disabled',
]
]);`

Please remove "id" from addField as its auto-increment. Pl try & let me know if this works.

**Comment by infiniteroles** [comment]
Great!!! It´s working perfect!!
Thank you very much!!

**Comment by jcastroa87** [comment]
Im sorry to lose this detail on code, thanks a lot @munjaldevelopment for your help.

As this is solve i will close the issue, but please feel free to re-open or create a new one if needed.

Cheers.
